test/t/struct_test.rb

```diff
@@ -28,6 +28,10 @@ def test_schema_can_be_derived_from_struct_with_default
     assert_equal(expected_schema, Job.schema)
   end
 
+  def test_serializer_returns_nested_hash_serializer
+    assert_kind_of(Typed::HashSerializer, City.serializer(:nested_hash))
```

Comment:
suggestion(non-blocking): We might want to add an attr reader for the serialize values boolean so we can verify that's getting set in these two tests.

Reply:
`Typed::HashSerializer` has a private `attr_reader` for `:should_serialize_values.` However, the abstract superclass `Serializer` doesn't have `should_serialize_values` as part of its interface since the `JSONSerializer` doesn't need it.

Should I test the boolean value on `Typed::HashSerializer` via `#send,` which is less than ideal?

Reply:
You can make that a public attr_reader on HashSerializer. It doesn't need to be on the Serializer interface, but can be publically accessible on HashSerializers.